script/cmake-build

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+#
+#  Copyright (c) 2020, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+set -euo pipefail
+
+readonly OT_PLATFORMS=(cc1352 cc2538 cc2650 cc2652 kw41z qpg6095 samr21 simulation posix)
+
+die()
+{
+    echo " ** ERROR: Openthread cmake doesn't support platform \"$1\""
+    exit 1
+}
+
+build()
+{
+    platform=$1
+    options=($2)
+    targets=($3)
+
+    mkdir -p build/${platform}
+    cd build/${platform}
+    cmake -GNinja ${options[*]} ../..
+    ninja ${targets[*]-}
+    cd ../..
+}
+
+main()
+{
+    if [[ $# == 0 ]]; then
+        >&2 echo "Please select a platform: ${OT_PLATFORMS[*]}"
+        # 128 for Invalid arguments
+        exit 128
+    fi
+
+    local targets=()
+    local platform="$1"
+    shift
+    local options=(
+        "-DOT_PLATFORM=${platform}"
+        "$@"
+    )
+
+    # Check if the platform supports cmake.
+    echo "${OT_PLATFORMS[@]}" | grep -wq "${platform}" || die ${platform}
+
+    if [[ "${platform}" != "posix" ]] && [[ "${platform}" != "simulation" ]]; then
+        options+=("-DCMAKE_TOOLCHAIN_FILE=examples/platforms/${platform}/arm-none-eabi.cmake")
+    fi
+
+    if [[ "${platform}" == "cc2650" ]]; then
+        targets+=("ot-cli-mtd" "ot-ncp-mtd")
+    fi
+
+    build "${platform}" "${options[*]}" "${targets[*]-}"
+}
+
+main "$@"
```

Comment:
nit: 1. Is it possible to add default options for vendor platforms and posix, simulation? 
2. Add usage for the script?

Reply:
1) will be convinient for posix and platforms. Furthermore, can we provide default options for all platforms?

Reply:
Will update the code.